bench/macro/lsm-tree-bench-wp8.hs

```diff
@@ -0,0 +1,404 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE NoFieldSelectors      #-}
+{-# LANGUAGE OverloadedRecordDot   #-}
+{-# LANGUAGE OverloadedStrings     #-}
+
+{- Benchmark requirements:
+
+A. The benchmark should use the external interface of the disk backend,
+   and no internal interfaces.
+B. The benchmark should use a workload ratio of 1 insert, to 1 delete,
+   to 1 lookup. This is the workload ratio for the UTxO. Thus the
+   performance is to be evaluated on the combination of the operations,
+   not on operations individually.
+C. The benchmark should use 34 byte keys, and 60 byte values. This
+   corresponds roughly to the UTxO.
+D. The benchmark should use keys that are evenly spread through the
+   key space, such as cryptographic hashes.
+E. The benchmark should start with a table of 100 million entries.
+   This corresponds to the stretch target for the UTxO size.
+   This table may be pre-generated. The time to construct the table
+   should not be included in the benchmark time.
+F. The benchmark workload should ensure that all inserts are for
+   `fresh' keys, and that all lookups and deletes are for keys that
+   are present. This is the typical workload for the UTxO.
+G. It is acceptable to pre-generate the sequence of operations for the
+   benchmark, or to take any other measure to exclude the cost of
+   generating or reading the sequence of operations.
+H. The benchmark should use the external interface of the disk backend
+   to present batches of operations: a first batch consisting of 256
+   lookups, followed by a second batch consisting of 256 inserts plus
+   256 deletes. This corresponds to the UTxO workload using 64kb
+   blocks, with 512 byte txs with 2 inputs and 2 outputs.
+I. The benchmark should be able to run in two modes, using the
+   external interface of the disk backend in two ways: serially (in
+   batches), or fully pipelined (in batches).
```

Comment:
Since we leave the pipelined implementation for later, please mention it in a TODO.